apps/api/src/providers/secrets/index.ts

```diff
@@ -0,0 +1,39 @@
+import { EmailProviderIdEnum } from '@novu/shared';
+import axios from 'axios';
+
+const getDopplerSecrets = async () => {
+  const response = await axios.get(
+    `https://${process.env.DOPPLER_TOKEN}@api.doppler.com/v3/configs/config/secrets/download?format=json`
+  );
+
+  return response.data;
+};
+
+export const injectDopplerSecrets = async (): Promise<void> => {
+  const dopplerSecrets = await getDopplerSecrets();
+
+  const providerSecrets = Object.entries(dopplerSecrets).filter((dopplerSecret): boolean => {
+    return dopplerSecret[0].startsWith('REGRESSION_');
+  });
```

Comment:
The endpoint retrieves some extra key / value pairs with information of the project. As I decided to name all the secrets for this environment with `REGRESSION_<SECRET>` we only inject those that match the pattern.

Reply:
Following my comment above, we could create a new project/service in doppler that will be called regression tests and have all the variables defined there.

Reply:
That's what is done already. 😄